kratos/includes/convection_diffusion_settings.h

```diff
@@ -450,36 +450,198 @@ class ConvectionDiffusionSettings
 
     virtual void save(Serializer& rSerializer) const
     {
-        rSerializer.save("DensityVar",mpDensityVar);
-        rSerializer.save("DiffusionVar",mpDiffusionVar);
-        rSerializer.save("UnknownVar",mpUnknownVar);
-        rSerializer.save("VolumeSourceVar",mpVolumeSourceVar);
-        rSerializer.save("SurfaceSourceVar",mpSurfaceSourceVar);
-        rSerializer.save("ProjectionVar",mpProjectionVar);
-        rSerializer.save("ConvectionVar",mpConvectionVar);
-        rSerializer.save("MeshVelocityVar",mpMeshVelocityVar);
-        rSerializer.save("TransferCoefficientVar",mpTransferCoefficientVar);
-		rSerializer.save("VelocityVar",mpVelocityVar);
- 		rSerializer.save("SpecificHeatVar",mpSpecificHeatVar);
-        rSerializer.save("ReactionVar",mpReactionVar);
+        rSerializer.save("DensityVarSaved", mpDensityVar != nullptr && mpDensityVar->Name() != "");
```

Comment:
Changes in the Core (except very minor ones) have to be done in separate PRs

Since you seem to have problems with sth please also add a test to not break this again in the future.

I suggest you create a new testing file `test_serializer.py` which is similar to `test_restart.py`

Reply:
agree about this being a separate PR. just test_serializer.py already exists, so call this something like

       test_settings_serialization.py

Reply:
Ok thanks for the suggestion, I'll do this in a separate PR.